drivers/scheduler/k8s/k8s.go

```diff
@@ -7517,3 +7528,29 @@ func ClusterVersion() (string, error) {
 	}
 	return strings.TrimLeft(ver.String(), "v"), nil
 }
+
+func createClonedStorageClassIfRequired(originalStorageClass *storageapi.StorageClass) (string, error) {
+	cloneStorageClass := originalStorageClass
+	clonedSCName := cloneStorageClass.Name + "-clone"
+	clonedSCobj, err := k8sStorage.GetStorageClass(clonedSCName)
+	if err == nil {
+		return clonedSCobj.Name, nil
+	} else if strings.Contains(err.Error(), fmt.Sprintf("storageclasses.storage.k8s.io \"%v\" not found", clonedSCName)) {
```

Comment:
nit for maintainability: 
Could you please provide a source of this error message format? 
If this code ever fails to function appropriately, future code maintainer should be able to find exactly where such a format is defined/changed.

Reply:
I printed the value of the err variable when the storage class is not present.

Reply:
Could you please find the source code of that error and include a link in the comment?

Reply:
Bump on this comment, I agree, hard error checks like this make me nervous for maintainability

Reply:
You can use `errors.IsNotFound(err)` I believe.

Reply:
Cannot use errors.IsNotFound(err) since it ss not able to catch the error.

Reply:
The source of error is k8s server but the error is returned to us by k8s client. 
Wdym by "Cannot use errors.IsNotFound(err) since it is not able to catch the error"? 

Using package "k8s.io/apimachinery/pkg/api/errors" and call `errors.IsNotFound(err)` to check error is widely used in torpedo in very similar use cases. 
For example: https://github.com/portworx/torpedo/blob/6b4160c1070b796bb10309d3a57cd2776b02736a/drivers/scheduler/k8s/k8s.go#L3844

Reply:
Discussed in the sync-up meeting: the confusion was trying to use the `IsNotFound` method outside of the Retry method. If we handle it inside the retry loop, it should work properly. (The issue was `errors.IsNotFound` doesn't work with the weird combined errors that the retry loop returns).